### PR TITLE
Astroport Farm

### PR DESCRIPTION
**Pull request by linkielink**
This PR enables the Farm UI for the Mars Protocol V2 outpost powered by Astroport Liquidity pools.
It includes
- Fetching and identification of Astroport LP Pool assets
- Listing whitelisted LP Pools in the Farm UI
- Fetching and displaying external incentives
- Deposit & Stake into and Unstake & Withdraw out of Astroport Pools
- Levered LP Farming with LP native tokens (swaping of borrowed tokens doesn't work until a SC update)
- Partial withdraws of staked LP positions
- Displaying staked LP positions as collateral 
- Updated Project dependencies and updated/enabled eslint v8

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  },
+  {
+    plugins: {
+      'react': eslintPluginReact,
+      'react-hooks': fixupPluginRules(eslintPluginReactHooks),
+    },
+  },
+  {
+    rules: {
+      // ...
```
Comment (bobthebuidlr) [inline eslint.config.mjs:35]:
```suggestion
```

File: src/components/earn/farm/Table/Columns/Apy.tsx
Comment (bobthebuidlr) [inline src/components/earn/farm/Table/Columns/Apy.tsx:1]: This file should then also be renamed  to VaultApy, as it's only used for that

File: src/components/common/TokenInput/index.tsx [hunk 2]
```diff
@@ -14,6 +14,7 @@ import useAssets from 'hooks/assets/useAssets'
 import useBaseAsset from 'hooks/assets/useBasetAsset'
 import { BNCoin } from 'types/classes/BNCoin'
 import { BN } from 'utils/helpers'
+import DoubleLogo from '../DoubleLogo'
```
Comment (bobthebuidlr) [inline src/components/common/TokenInput/index.tsx:17]: path

**Review by bobthebuidlr — CHANGES_REQUESTED** [review]
My general remarks:

1. Use a separated folder the table/columns for `Vault` and `Farm`. Especially the `Columns` folder is super unorganized and has both common and individual components.

2. 90% of the code lines in folder `Modals/Farm` is a duplicated from `Modals/Vault`. DRY: please create common components 

3. The naming of `Farm` feels confusing and incorrect. Farming is the overarching term for getting yield over your assets. The SC termed it `StakedAstroLp` for a reason. Right now you are using Farm in two different ways: as a general term for any yield strategies AND as the term for staking in LPs. Aside from that you still do refer a lot to either stakedLP and/or LP, which makes it even more confusing to follow along.

File: src/store/slices/broadcast.ts [hunk 3]
```diff
@@ -409,6 +409,88 @@ export default function createBroadcastSlice(
...
+
+        actions.push({
+          unstake_astro_lp: {
+            lp_token: coin,
+          },
+        })
+        actions.push({
+          withdraw_liquidity: {
+            lp_token: coin,
+            slippage: '0',
```
[the remark this answers is not in the document]
Comment (linkielink) [reply src/store/slices/broadcast.ts:433]: Yes, withdrawing liquidity from Astroport doesn't include any slippage. None the less is slippage mandatory for `withdraw_liquidity`

File: src/hooks/assets/useAssetsNoOraclePrices.ts
[the remark this answers is not in the document]
Comment (linkielink) [reply src/hooks/assets/useAssetsNoOraclePrices.ts:1]: Fetched assets from Astroport can have a price, not an oracle price, but a TWAP price. All assets that have Oracle prices get overwritten by the Oracle prices. All that don't update their price on refetch.

**Comment by linkielink** [comment]
> My general remarks:
> 
> 1. Use a separated folder the table/columns for `Vault` and `Farm`. Especially the `Columns` folder is super unorganized and has both common and individual components.
> 2. 90% of the code lines in folder `Modals/Farm` is a duplicated from `Modals/Vault`. DRY: please create common components
> 3. The naming of `Farm` feels confusing and incorrect. Farming is the overarching term for getting yield over your assets. The SC termed it `StakedAstroLp` for a reason. Right now you are using Farm in two different ways: as a general term for any yield strategies AND as the term for staking in LPs. Aside from that you still do refer a lot to either stakedLP and/or LP, which makes it even more confusing to follow along.

I renamed and restructured the folders, types and values to represent `astroLp` instead of `farm`.
The `AstroLpModal` is pretty similiar to the `VaultModal`, however I plan on restructuring and refactoring those after the RewardsCenter update. 

Please check if the naming is now less confusing and if the folder structure is better to read

**Review by bobthebuidlr — APPROVED** [review]

**Pull request merged**